File: frontend/src/compartido/dominio/eventos.ts
/**
 * eventos.ts — Tipos de eventos del sistema
 */

export interface RedActualizada {
  tipo: 'RedActualizada';
  redId: string;
}

export interface SimulacionIniciada {
  tipo: 'SimulacionIniciada';
  redId: string;
}

export interface ErrorCalculo {
  tipo: 'ErrorCalculo';
  mensaje: string;
}

export type EventoSistema = RedActualizada | SimulacionIniciada | ErrorCalculo;

File: backend/compartido/dominio/eventos.py
"""
eventos.py — Eventos del sistema entre componentes
"""

from dataclasses import dataclass


@dataclass
class RedActualizada:
    tipo: str = 'RedActualizada'
    redId: str = ''


@dataclass
class SimulacionIniciada:
    tipo: str = 'SimulacionIniciada'
    redId: str = ''


@dataclass
class ErrorCalculo:
    tipo: str = 'ErrorCalculo'
    mensaje: str = ''